molecule-runtime/src/macosMain/kotlin/app/cash/molecule/DisplayLinkClock.kt

```diff
@@ -0,0 +1,54 @@
+package app.cash.molecule
+
+import androidx.compose.runtime.BroadcastFrameClock
+import androidx.compose.runtime.MonotonicFrameClock
+import kotlinx.cinterop.alloc
+import kotlinx.cinterop.allocPointerTo
+import kotlinx.cinterop.nativeHeap
+import kotlinx.cinterop.ptr
+import kotlinx.cinterop.staticCFunction
+import kotlinx.cinterop.value
+import platform.CoreVideo.CVDisplayLinkCreateWithActiveCGDisplays
+import platform.CoreVideo.CVDisplayLinkRef
+import platform.CoreVideo.CVDisplayLinkRefVar
+import platform.CoreVideo.CVDisplayLinkSetOutputCallback
+import platform.CoreVideo.CVDisplayLinkStart
+import platform.CoreVideo.CVDisplayLinkStop
+import platform.CoreVideo.kCVReturnSuccess
+
+public actual object DisplayLinkClock : MonotonicFrameClock {
+
+  private val clock = BroadcastFrameClock {
+    // One or more awaiters have appeared. Start the DisplayLink clock callback so that awaiters
+    // get dispatched on the next available frame.
+    checkDisplayLink(CVDisplayLinkStart(displayLink.value))
+  }
```

Comment:
I'm curious to hear opinions on this approach. Rather than making these clocks classes, I decided to make them singleton objects that only register their callbacks when a frame is awaiting dispatch.

Reply:
The Android one behaves the same way, although it's not a singleton.

Reply:
Is there any benefit I might not be thinking for classes over a singleton here?

Reply:
Nope